mautrix_facebook/portal.py

```diff
@@ -496,14 +499,22 @@ def optional_send_lock(self, user_id: str) -> Union[asyncio.Lock, FakeLock]:
         if message.sticker:
             event_ids = [await self._handle_facebook_sticker(intent, message.sticker,
                                                              message.reply_to_id)]
-        elif len(message.attachments) > 0:
-            event_ids = await asyncio.gather(
-                *[self._handle_facebook_attachment(intent, attachment, message.reply_to_id)
-                  for attachment in message.attachments])
-            event_ids = [event_id for event_id in event_ids if event_id]
-        if not event_ids and message.text:
-            event_ids = [await self._handle_facebook_text(intent, message)]
         else:
+            if message.text:
+                event_ids = [await self._handle_facebook_text(intent, message)]
+
+            if len(message.attachments) > 0:
+                # Get URLs in the original message to prevent sending unnecessary ShareAttachments
+                urls = self._urlRegex.findall(message.text)
```

Comment:
Does this really need to split into a list with a regex? i.e. can't it just do `if url not in message.text: <handle attachment>`. If `url` has a trailing slash, just remove that first. If `message.text` has a trailing slash, it'll still work for substring finding.

Reply:
So simple I didn't thought of it, the regex is totally overkill.